x-pack/plugins/cloud_security_posture/public/pages/vulnerabilities/vulnerabilities.tsx

```diff
@@ -80,20 +81,23 @@ const VulnerabilitiesContent = ({ dataView }: { dataView: DataView }) => {
     enabled: !queryError,
   });
 
-  const [isVulnerabilityDetailFlyoutVisible, setIsVulnerabilityDetailFlyoutVisible] =
-    useState(false);
+  const invalidIndex = -1;
+  const selectedVulnerability = data?.page[urlQuery.vulnerabilityIndex];
 
-  const [vulnerability, setVulnerability] = useState<VulnerabilityRecord>();
-
-  const showFlyout = (vulnerabilityRecord: VulnerabilityRecord) => {
-    setIsVulnerabilityDetailFlyoutVisible(true);
-    setVulnerability(vulnerabilityRecord);
+  const onCloseFlyout = () => {
+    setUrlQuery({
+      vulnerabilityIndex: invalidIndex,
+    });
   };
 
-  const hideFlyout = () => {
-    setIsVulnerabilityDetailFlyoutVisible(false);
-    setVulnerability(undefined);
-  };
+  const onOpenFlyout = useCallback(
+    (rowIndex: number) => {
+      setUrlQuery({
+        vulnerabilityIndex: rowIndex,
```

Comment:
consider using an `id` instead of an index. indexes can change when the data is modified or sorted, ids will stay the same. will make pagination harder though